tests/Test.Microsoft.Identity.LabInfrastructure/LabAuthenticationHelper.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Test.Microsoft.Identity.LabInfrastructure
+{
+    public static class LabAuthenticationHelper
+    {
+        /// <summary>
+        /// Token cache used by the test infrastructure when authenticating against KeyVault
+        /// </summary>
+        /// <remarks>We aren't using the default cache to make sure the tokens used by this
+        /// test infrastructure can't end up in the cache being used by the tests (the UI-less
+        /// Desktop test app runs in the same AppDomain as the infrastructure and uses the
+        /// default cache).</remarks>
+        private readonly static TokenCache keyVaultTokenCache = new TokenCache();
+
+        private const string LabAccessConfidentialClientId = "16dab2ba-145d-4b1b-8569-bf4b9aed4dc8";
+        private const string LabAccessPublicClientId = "3c1e0e0d-b742-45ba-a35e-01c664e14b16";
+        private const string LabAccessThumbPrint = "3051A5BE699BC4596EE47E9FEBBF48DBA85BE67B";
+        private const string DataFileName = "data.txt";
+        private static LabAccessAuthenticationType s_defaultAuthType = LabAccessAuthenticationType.ClientCertificate;
+        private static string s_secret;
+        private static ClientAssertionCertificate _assertionCert;
+        private static AuthenticationResult _authResult;
+
+        static LabAuthenticationHelper()
+        {
+            //The data.txt is a place holder for the keyvault secret. It will only be written to during build time when testing appcenter.
+            //After the tests are finished in appcenter, the file will be deleted from the appcenter servers.
+            //The file will then be deleted locally Via VSTS task.
+            if (File.Exists(DataFileName))
```

Comment:
Do we still do this ? I don't think so, but may be wrong...

Reply:
no idea. i'll take a look tomorrow.

Reply:
this is needed.